[lldb] Introduce an ImportedDeclaration and [lldb] Handle @_originallyDefinedIn [main]

augusto2112 opened the pull request:
This PR contains two commits:

[lldb] Introduce an ImportedDeclaration

Introduce a debug info independent type, ImportedDeclaration, which is
analogous to DWARF's DW_AT_imported_declaration, as well as a way to
search them by name.

--------------------------------------------------------------------------

[lldb] Handle @_originallyDefinedIn

Types annotated with @_originallyDefinedIn don't live in the module
listed in their mangled name. To account for this, the compiler emits a
DW_TAG_imported_declaration for those types under the name of the
swiftmodule these types can be found at. This patch implements the logic
required to produce the mangled name that can be used in type
reconstruction to successfully find these types.

rdar://137146961

augusto2112 committed:
[lldb] Introduce an ImportedDeclaration

Introduce a debug info independent type, ImportedDeclaration, which is
analogous to DWARF's DW_AT_imported_declaration, as well as a way to
search them by name.

augusto2112 commented:
https://github.com/swiftlang/swift/pull/78104
@swift-ci test

augusto2112 commented:
https://github.com/swiftlang/swift/pull/78104
@swift-ci test

adrian-prantl commented on lldb/include/lldb/Symbol/ImportedDeclaration.h:
```diff
@@ -0,0 +1,34 @@
+  SymbolFile *m_symbol_file = nullptr;
+};
+
+} // namespace lldb_private
```
Please make sure to propose this interface upstream and carry forward all changes coming out of that discussion.

augusto2112 replied:
Yes I plan the upstream the commit with the `ImportedDeclaration`.

augusto2112 commented:
https://github.com/swiftlang/swift/pull/78104
@swift-ci test

adrian-prantl commented on lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:
```diff
@@ -157,6 +158,261 @@ TypeSystemSwiftTypeRef::CanonicalizeSugar(swift::Demangle::Demangler &dem,
+  return mangling;
+}
+
+static NodePointer CloneNode(Demangler &dem, NodePointer node,
```
Doxygen comment that explains what set_children does?

adrian-prantl commented on lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:
```diff
@@ -157,6 +158,261 @@ TypeSystemSwiftTypeRef::CanonicalizeSugar(swift::Demangle::Demangler &dem,
+
+static NodePointer CloneNode(Demangler &dem, NodePointer node,
+                             bool set_children = true) {
+  NodePointer clone;
```
Should this be in the demangle helper functions file?

adrian-prantl commented on lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:
```diff
@@ -157,6 +158,261 @@ TypeSystemSwiftTypeRef::CanonicalizeSugar(swift::Demangle::Demangler &dem,
+  return clone;
+}
+
+NodePointer TypeSystemSwiftTypeRef::TransformTupleElements(
```
How is this different from the `Transform()` function?

augusto2112 replied:
This only applies the transformation function to the elements of a tuple. It _could_ be implemented in terms of `Transform`, but I think it'd suffer in terms of legibility, since the implementation would be less straightforward.

augusto2112 commented:
https://github.com/swiftlang/swift/pull/78104
@swift-ci test

augusto2112 commented:
https://github.com/swiftlang/swift/pull/78104
@swift-ci test

augusto2112 commented:
https://github.com/swiftlang/swift/pull/78104
@swift-ci test

adrian-prantl commented on lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:
```diff
@@ -157,6 +158,244 @@ TypeSystemSwiftTypeRef::CanonicalizeSugar(swift::Demangle::Demangler &dem,
+    const llvm::StringMap<llvm::StringRef> &module_name_map) {
+  swift::Demangle::Demangler dem;
+  auto *node = dem.demangleSymbol(mangled_name);
+  auto *adjusted_node = TypeSystemSwiftTypeRef::Transform(
```
Transform checks for nullptr, right?

augusto2112 replied:
Yes

adrian-prantl commented on lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:
```diff
@@ -157,6 +158,244 @@ TypeSystemSwiftTypeRef::CanonicalizeSugar(swift::Demangle::Demangler &dem,
+    // If the node has children this is not the module node yet, so recursively
+    // examine the first child.
+    if (node->hasChildren()) {
+      NodePointer transformed = recurse_to_module_node(node->getFirstChild());
```
Why is it okay to only dive into the first child?

augusto2112 replied:
Because a nominal type's module will the at position 0 at some depth.

For example, a simple struct:
```
struct A {}
```

Will have mangling tree:
```
kind=Global
  kind=TypeMangling
    kind=Type
      kind=Structure
        kind=Module, text="Module"
        kind=Identifier, text="A"
```

A bound generic will still have the module at position 0, but deeper in the tree:

```
kind=Global
  kind=TypeMangling
    kind=Type
      kind=BoundGenericStructure
        kind=Type
          kind=Structure
            kind=Module, text="Swift"
            kind=Identifier, text="Array"
        // TypeList subtree here
```

The module node can be at some arbitrary depth since you can have arbitrarily nested types.

adrian-prantl commented on lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:
```diff
@@ -157,6 +158,244 @@ TypeSystemSwiftTypeRef::CanonicalizeSugar(swift::Demangle::Demangler &dem,
+  }
+
+  bool failed = false;
+  std::function<NodePointer(NodePointer)> recurse_to_module_node =
```
I find it confusing that a function called recurse_to_module_node actually modifies something.

adrian-prantl commented on lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:
```diff
@@ -157,6 +158,244 @@ TypeSystemSwiftTypeRef::CanonicalizeSugar(swift::Demangle::Demangler &dem,
+      // same, and we don't want to accidentally mutate some other part of the
+      // tree.
+      NodePointer clone = swift_demangle::CloneNode(dem, node);
+      clone->replaceChild(0, transformed);
```
Is this is building the tree back up starting at the leaves. Can you give a concrete example where it would not be correct to mutate the tree in-place?
Or is it to guarantee against a completely unrelated demangle tree owned by the same Demangler being modified?

augusto2112 replied:
Demangle trees can share subtrees, I assume for performance reasons. I found out about this with the test I added, I'll check how mutating it in place fails exactly.

adrian-prantl commented on lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.h:
```diff
@@ -374,6 +374,40 @@ class TypeSystemSwiftTypeRef : public TypeSystemSwift {
+  /// Transforms the module name in the mangled type name using module_name_map
+  /// as the mapping source.
+  static swift::Demangle::ManglingErrorOr<std::string>
+  TransformModuleName(llvm::StringRef mangled_name,
```
This family of static functions should be unit tested in TestSwiftASTContext.cpp

adrian-prantl commented on lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:
```diff
@@ -157,6 +158,244 @@ TypeSystemSwiftTypeRef::CanonicalizeSugar(swift::Demangle::Demangler &dem,
+
+    NodePointer transformed_generic_types =
+        TypeSystemSwiftTypeRef::TransformBoundGenericTypes(
+            dem, transformed_module, transform_all_module_nodes);
```
I'm not quite sold on the architecture of this algorithm yet. Why isn't this a single invocation of transform, that switches over the node type dispatches to local transformations?

augusto2112 committed:
[lldb] Handle @_originallyDefinedIn

Types annotated with @_originallyDefinedIn don't live in the module
listed in their mangled name. To account for this, the compiler emits a
DW_TAG_imported_declaration for those types under the name of the
swiftmodule these types can be found at. This patch implements the logic
required to produce the mangled name that can be used in type
reconstruction to successfully find these types.

rdar://137146961

adrian-prantl reviewed (approved):
Nice! This addresses all my previous concerns.

augusto2112 commented:
https://github.com/swiftlang/swift/pull/78104
@swift-ci test

augusto2112 commented:
https://github.com/swiftlang/swift/pull/78104
@swift-ci test

augusto2112 commented:
https://github.com/swiftlang/swift/pull/78104
@swift-ci test

augusto2112 commented:
Unrelated Windows failure:

```
"T:\5\bin\swiftc.exe -target x86_64-unknown-windows-msvc -j 36 -num-threads 36 -c -DFoundationEssentials_EXPORTS -parse-as-library -emit-module -emit-module-path swift\FoundationEssentials.swiftmodule -module-name FoundationEssentials -module-link-name FoundationEssentials -vfsoverlay T:/101/stdlib/windows-vfs-overlay.yaml -strict-implicit-module-context -Xcc -Xclang -Xcc -fbuiltin-headers-in-system-modules -resource-dir T:/101/lib/swift -L T:/101/lib/swift/windows -gnone -Xlinker /INCREMENTAL:NO -Xlinker /OPT:REF -Xlinker /OPT:ICF -O -libc MD -incremental -plugin-path T:/2/bin -Xfrontend -enable-experimental-feature -Xfrontend VariadicGenerics -Xfrontend -enable-experimental-feature -Xfrontend AccessLevelOnImport -Xfrontend -enable-experimental-feature -Xfrontend StrictConcurrency -Xfrontend -enable-upcoming-feature -Xfrontend InferSendableFromCaptures -Xfrontend -enable-experimental-feature -Xfrontend "AvailabilityMacro=FoundationPreview 0.1:macOS 13.3, iOS 16.4, tvOS 16.4, watchOS 9.4" -Xfrontend -enable-experimental-feature -Xfrontend "AvailabilityMacro=FoundationPredicate 0.1:macOS 14, iOS 17, tvOS 17, watchOS 10" -Xfrontend -enable-experimental-feature -Xfrontend "AvailabilityMacro=FoundationPredicateRegex 0.1:macOS 15, iOS 18, tvOS 18, watchOS 11" -Xfrontend -enable-experimental-feature -Xfrontend "AvailabilityMacro=FoundationPreview 0.2:macOS 13.3, iOS 16.4, tvOS 16.4, watchOS 9.4" -Xfrontend -enable-experimental-feature -Xfrontend "AvailabilityMacro=FoundationPredicate 0.2:macOS 14, iOS 17, tvOS 17, watchOS 10" -Xfrontend -enable-experimental-feature -Xfrontend "AvailabilityMacro=FoundationPredicateRegex 0.2:macOS 15, iOS 18, tvOS 18, watchOS 11" -Xfrontend -enable-experimental-feature -Xfrontend "AvailabilityMacro=FoundationPreview 0.3:macOS 13.3, iOS 16.4, tvOS 16.4, watchOS 9.4" -Xfrontend -enable-experimental-feature -Xfrontend "AvailabilityMacro=FoundationPredicate 0.3:macOS 14, iOS 17, tvOS 17, watchOS 10" -Xfrontend -enable-experimental-feature -Xfrontend "AvailabilityMacro=FoundationPredicateRegex 0.3:macOS 15, iOS 18, tvOS 18, watchOS 11" -Xfrontend -enable-experimental-feature -Xfrontend "AvailabilityMacro=FoundationPreview 0.4:macOS 13.3, iOS 16.4, tvOS 16.4, watchOS 9.4" -Xfrontend -enable-experimental-feature -Xfrontend "AvailabilityMacro=FoundationPredicate 0.4:macOS 14, iOS 17, tvOS 17, watchOS 10" -Xfrontend -enable-experimental-feature -Xfrontend "AvailabilityMacro=FoundationPredicateRegex 0.4:macOS 15, iOS 18, tvOS 18, watchOS 11" -Xfrontend -enable-experimental-feature -Xfrontend "AvailabilityMacro=FoundationPreview 6.0.2:macOS 13.3, iOS 16.4, tvOS 16.4, watchOS 9.4" -Xfrontend -enable-experimental-feature -Xfrontend "AvailabilityMacro=FoundationPredicate 6.0.2:macOS 14, iOS 17, tvOS 17, watchOS 10" -Xfrontend -enable-experimental-feature -Xfrontend "AvailabilityMacro=FoundationPredicateRegex 6.0.2:macOS 15, iOS 18, tvOS 18, watchOS 11" -Xfrontend -enable-experimental-feature -Xfrontend "AvailabilityMacro=FoundationPreview 6.2:macOS 13.3, iOS 16.4, tvOS 16.4, watchOS 9.4" -Xfrontend -enable-experimental-feature -Xfrontend "AvailabilityMacro=FoundationPredicate 6.2:macOS 14, iOS 17, tvOS 17, watchOS 10" -Xfrontend -enable-experimental-feature -Xfrontend "AvailabilityMacro=FoundationPredicateRegex 6.2:macOS 15, iOS 18, tvOS 18, watchOS 11" -package-name SwiftFoundation -Xcc -fmodule-map-file=C:/Users/swift-ci/jenkins/workspace/apple-llvm-project-pull-request-windows/swift-foundation/Sources/_FoundationCShims/include/module.modulemap -output-file-map _deps\swiftfoundation-build\Sources\FoundationEssentials\CMakeFiles\FoundationEssentials.dir\Release\output-file-map.json -I T:\103\swift -I C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\_FoundationCShims\include C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\_ThreadLocal.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Bundle+Stub.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Codable.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\CodableUtilities.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\CodableWithConfiguration.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\ComparisonResult.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Date.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\DateInterval.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\FoundationEssentials.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\IndexPath.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\LockedState.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Logging.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\OutputBuffer.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Platform.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\SortComparator.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\UUID_Wrappers.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\UUID.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\WASILibc+Extensions.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\WinSDK+Extensions.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\AttributeContainer.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\AttributeScope.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\AttributedString+AttributeTransformation.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\AttributedString+CharacterView.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\AttributedString+Guts.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\AttributedString+Runs+AttributeSlices.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\AttributedString+Runs+Run.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\AttributedString+Runs.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\AttributedString+UnicodeScalarView.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\AttributedString+UTF8View.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\AttributedString+UTF16View.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\AttributedString+_InternalRun.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\AttributedString+_InternalRuns.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\AttributedString+_InternalRunsSlice.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\AttributedString.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\AttributedStringAttribute.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\AttributedStringAttributeConstrainingBehavior.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\AttributedStringAttributeStorage.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\AttributedStringCodable.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\AttributedStringProtocol.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\AttributedSubstring.swift "C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\Collection Stdlib Defaults.swift" C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\Conversion.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\FoundationAttributes.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\AttributedString\String.Index+ABI.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Calendar\Calendar.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Calendar\Calendar_Autoupdating.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Calendar\Calendar_Cache.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Calendar\Calendar_Enumerate.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Calendar\Calendar_Gregorian.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Calendar\Calendar_Protocol.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Calendar\Calendar_Recurrence.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Calendar\Date+FormatStyle.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Calendar\Date+Locale.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Calendar\DateComponents.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Calendar\RecurrenceRule.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Data\Collections+DataProtocol.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Data\ContiguousBytes.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Data\Data+Base64.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Data\Data+Error.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Data\Data+Reading.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Data\Data+Stub.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Data\Data+Writing.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Data\Data.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Data\DataProtocol.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Data\Pointers+DataProtocol.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Decimal\Decimal.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Decimal\Decimal+Conformances.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Decimal\Decimal+Math.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Decimal\Decimal+Compatibility.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Error\CocoaError+FilePath.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Error\CocoaError.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Error\ErrorCodes+POSIX.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Error\ErrorCodes+Win32.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Error\ErrorCodes.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\FileManager\SearchPaths\FileManager+DarwinSearchPaths.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\FileManager\SearchPaths\FileManager+SearchPaths.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\FileManager\SearchPaths\FileManager+WindowsSearchPaths.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\FileManager\SearchPaths\FileManager+XDGSearchPaths.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\FileManager\FileManager+Basics.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\FileManager\FileManager+Bridge.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\FileManager\FileManager+Directories.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\FileManager\FileManager+Files.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\FileManager\FileManager+SymbolicLinks.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\FileManager\FileManager+Utilities.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\FileManager\FileOperations+Enumeration.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\FileManager\FileOperations.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\FileManager\SwiftFileManager.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\FileManager\SwiftFileManagerDelegate.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Formatting\BinaryInteger+NumericStringRepresentation.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Formatting\Date+ISO8601FormatStyle.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Formatting\DiscreteFormatStyle.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Formatting\FormatParsingUtilities.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Formatting\FormatStyle.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Formatting\FormatterCache.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Formatting\ParseStrategy.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Formatting\ParseableFormatStyle.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\JSON\BufferView.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\JSON\BufferViewCompatibility.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\JSON\BufferViewIndex.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\JSON\BufferViewIterator.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\JSON\JSON5Scanner.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\JSON\JSONDecoder.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\JSON\JSONEncoder.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\JSON\JSONScanner.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\JSON\JSONWriter.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Locale\Locale+Components.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Locale\Locale+Language.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Locale\Locale.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Locale\Locale_Autoupdating.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Locale\Locale_Cache.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Locale\Locale_Notifications.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Locale\Locale_Preferences.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Locale\Locale_Protocol.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Locale\Locale_Unlocalized.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Archiving\EncodingContainers+PredicateExpression.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Archiving\Predicate+Codable.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Archiving\PredicateCodableConfiguration.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Archiving\PredicateExpressionConstruction.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Expressions\Aggregate.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Expressions\Arithmetic.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Expressions\Collection.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Expressions\Comparison.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Expressions\Conditional.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Expressions\Conjunction.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Expressions\Dictionary.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Expressions\Disjunction.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Expressions\Division.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Expressions\Equality.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Expressions\ExpressionEvaluation.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Expressions\Filter.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Expressions\Inequality.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Expressions\Negation.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Expressions\Optional.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Expressions\PredicateEvaluation.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Expressions\Range.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Expressions\Regex.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Expressions\Sequence.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Expressions\StringComparison.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Expressions\Types.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Expressions\UnaryMinus.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Expression.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\KeyPath+Inspection.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\NSPredicateConversion.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Predicate+Description.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\Predicate.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\PredicateBindings.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\Predicate\PredicateExpression.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\ProcessInfo\ProcessInfo+API.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\ProcessInfo\ProcessInfo+ObjC.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\ProcessInfo\ProcessInfo.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\PropertyList\BPlistDecodingFormat.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\PropertyList\BPlistEncodingFormat.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\PropertyList\BPlistScanner.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\PropertyList\OpenStepPlist.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\PropertyList\PlistDecoder.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\PropertyList\PlistDecoderGeneric.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\PropertyList\PlistDictionaryDecoder.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\PropertyList\PlistDictionaryEncoder.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\PropertyList\PlistEncoder.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\PropertyList\PlistEncoderGeneric.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\PropertyList\XMLPlistDecodingFormat.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\PropertyList\XMLPlistEncodingFormat.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\PropertyList\XMLPlistScanner.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\String\BidirectionalCollection.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\String\BuiltInUnicodeScalarSet.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\String\RegexPatternCache.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\String\String+Comparison.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\String\String+Encoding.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\String\String+EndianAdaptorSequence.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\String\String+Essentials.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\String\String+IO.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\String\String+Internals.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\String\String+Path.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\String\StringBlocks.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\String\StringProtocol+Essentials.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\String\StringProtocol+Stub.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\String\UnicodeScalar.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\TimeZone\TimeZone.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\TimeZone\TimeZone_Autoupdating.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\TimeZone\TimeZone_Cache.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\TimeZone\TimeZone_GMT.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\TimeZone\TimeZone_Protocol.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\URL\URL.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\URL\URLComponents.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\URL\URLComponents_ObjC.swift C:\Users\swift-ci\jenkins\workspace\apple-llvm-project-pull-request-windows\swift-foundation\Sources\FoundationEssentials\URL\URLParser.swift"
```
